diff --git a/CMakeLists.txt b/CMakeLists.txt
index 212aa9c..76c05ee 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -270,11 +270,9 @@ if(MSVC)
   # needed to compile protobuf
   set(_gRPC_C_CXX_FLAGS "${_gRPC_C_CXX_FLAGS} /wd4065 /wd4506")
   # TODO(jtattermusch): revisit warnings that were silenced as part of upgrade to protobuf3.6.0
-  set(_gRPC_C_CXX_FLAGS "${_gRPC_C_CXX_FLAGS} /wd4200 /wd4291 /wd4244")
-  # TODO(jtattermusch): revisit C4267 occurrences throughout the code
-  set(_gRPC_C_CXX_FLAGS "${_gRPC_C_CXX_FLAGS} /wd4267")
+  set(_gRPC_C_CXX_FLAGS "${_gRPC_C_CXX_FLAGS} /wd4200 /wd4291")
   # TODO(jtattermusch): needed to build boringssl with VS2017, revisit later
-  set(_gRPC_C_CXX_FLAGS "${_gRPC_C_CXX_FLAGS} /wd4987 /wd4774 /wd4819 /wd4996 /wd4619")
+  set(_gRPC_C_CXX_FLAGS "${_gRPC_C_CXX_FLAGS} /wd4987 /wd4774 /wd4819 /wd4619")
   if(_gRPC_PLATFORM_UWP)
     add_definitions(-DGRPC_ARES=0)
   endif()
